src/config.rs

```diff
@@ -60,6 +60,11 @@ pub struct Config {
     /// Default: 1
     pub format_version: Version,
 
+    /// Layout of data in log file.
+    ///
+    /// Default: "no-alignment"
+    pub format_data_layout: DataLayout,
```

Comment:
Why adding a configuration? Everything should be read from file header.

Reply:
Yes, it's a necessary configuration to user to use which format to dump `LogBatch`s into log files.
For `write`:
* If `no-alignment`, the log content will be dumped into log files directly, without alignment or fragments.
* If `align-with-integration`, the log content will be aligned with default block size, and dumped into log files. [For future DIO]
* If `align-with-fragments`, the `LogBatch` will be aligned and divided into several parts, typed with `Full || First || Middle || Last` into the log file.

For `read`:
This context in the header will  be parsed and make the `Reader` how to read the context from the file, no aligned format, aligned format in integration or aligned format in fragments.

Reply:
> Yes, it's a necessary configuration to user to use which format to dump `LogBatch`s into log files. For `write`:
> 
> * If `no-alignment`, the log content will be dumped into log files directly, without alignment or fragments.
> * If `align-with-integration`, the log content will be aligned with default block size, and dumped into log files. [For future DIO]
> * If `align-with-fragments`, the `LogBatch` will be aligned and divided into several parts, typed with `Full || First || Middle || Last` into the log file.
> 
> For `read`: This context in the header will be parsed and make the `Reader` how to read the context from the file, no aligned format, aligned format in integration or aligned format in fragments.
> 
> Meanwhile, this configuration is a compatible design to make the parsing strategy, when `recovery` or `get_entry`, on current format compatible to the future fragmented records in log file.

Ignore it. This configuration is over-designed. I'll change it.